### ISSUE TITLE
Adding a relationship

### ISSUE THREAD

**Issue by brunobl** [vapor/api-template#29]
Template should be expanded with some kind of relationship. Either User->Posts, or Post->Comments, so as to explain how to handle this.

Documentation currently isn't very clear, and I didn't manage to get help on Slack either, so it seems others are struggling about this too.
This would also explain best practice, and how to use Children<,> foreignId() and so on.

**Comment by tanner0101** [comment]
Could you explain which parts of the documentation aren't clear? It has descriptions and code samples for each type of relation: https://docs.vapor.codes/2.0/fluent/relations/

The API template should be a starting point for new projects, not an example project. Ideally examples/samples/etc would go into the documentation (this should be an issue opened on vapor/documentation).

**Comment by brunobl** [comment]
I have seen several people asking the same question on Slack, and nobody
ever answer, so obviously I'm not alone not figuring out how to make
everything works together. Basically we need a full sample code for this
basic topic. I managed to make it work last year for Vapor1 when I tested
just for fun, but I'm no longer able to make it work for Vapor2.


1. JSON: I've been unable to implement

convenience init(json: JSON) throws

in the case there is a foreign id which is Identifier? type. Nobody can
help me too. So documentation isn't good enough.
https://docs.vapor.codes/2.0/json/overview/

2. I have seen another issue closed about resources. There is no sample
code in the documentation about how to use each kind of resource or what
they do. Only index: and show:
api-template doesn't implement all of them
You can't close so quickly saying documentation is good enough when it isn't

3. It's also a matter of best practice. The few sample codes aren't using
the modern Vapor2 features. If documentation and template don't show what
to do, people are going to rely on wrong things.

**Comment by tanner0101** [comment]
@brunobl I appreciate your comments here however they are too vague to address directly. Please create issue reports for any individual, specific problems you have and we can tag, assign, and address them appropriately.

**Comment by brunobl** [comment]
You have already closed what I wrote in point 2 (someone else opened an
individual issue about this).
You have closed this issue #29 without even letting me answer first.

If you do the same for everything documentation related, why should we
spend time on this...
I managed to find by myself what I wanted anyway :-(
I yesterday read an article that poor documentation was one of the most
common reason of opensource project failure. It was striking me, because it
was exactly what I meant. A project need proper documentation with proper
detailed tutorials and fully working source code samples, not just short
snippets which don't work by themselves. Maybe it would be good to slow
down on development a few weeks and focus on better documentation...
